src/js/components/Table/Table.tsx

```diff
@@ -75,40 +81,37 @@ class Table extends React.PureComponent<any, any> {
     placeHolder: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
   };
 
-  state = {
-    items: [],
-    sortByColumn: "",
-    sortByReverse: false
-  };
-
-  componentWillMount() {
-    const { rows, defaultSort } = this.props;
+  constructor(props) {
+    super(props);
+    const { defaultSort } = this.props;
 
     // Establish initial sortDirection by checking for '-' value
     const sortDirection = defaultSort.charAt(0) === "-" ? true : false;
     const sortName = sortDirection ? defaultSort.substr(1) : defaultSort;
 
-    this.setTableState(rows, sortName, sortDirection);
+    const state = Table.getTableState(props, sortName, sortDirection);
+    this.state = { ...state };
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { sortByColumn, sortByReverse } = this.state;
+  static getDerivedStateFromProps(props, state) {
+    const { sortByColumn, sortByReverse } = state;
 
-    this.setTableState(nextProps.rows, sortByColumn, sortByReverse);
+    return Table.getTableState(props, sortByColumn, sortByReverse);
   }
 
-  setTableState = (rows, sortName, sortDirection) => {
+  private static getTableState(props, sortName, sortDirection) {
+    const { rows } = props;
     const items = setUniqueId(rows);
 
     const tableState = sortName
-      ? this.sortItems(items, sortName, sortDirection)
+      ? Table.sortItems(props, items, sortName, sortDirection)
       : { items };
 
-    this.setState({ ...tableState });
-  };
+    return tableState;
+  }
 
-  sortItems = (newItems, sortByColumn, sortByReverse) => {
-    const { columns } = this.props;
+  static sortItems = (props, newItems, sortByColumn, sortByReverse) => {
```

Comment:
I'm curious about which methods are private. What's the thinking behind private getTableState and not sortItems?

Reply:
A great question, I made new methods private and left the others as is (for now). Unfortunately, we have a public library using semantic versioning, so it'd be unwise to start making public methods private without a major version bump (people might be using those methods!)

Ultimately, the rules for API design that I try to follow are as follows:
1. New methods should be private unless they otherwise need a public modifier.
2. You can always make private methods public later (that wont break anything)
3. Making public methods private requires significant overhead (or a PSA that things may break).

That being said, since we're moving sortItems to a static modifier, I guess that technically is a breaking change, so it might make sense to go ahead and make it private as well. The only downside is that if someone is using the public sort items, it's not hard to simply change the callsite to a static call, but if its private, they can no longer use it (they'll have to write their own).

Reply:
Great notes @bradcypert I wonder if we should document some of these as a wiki page or in the readme.